collections/deep_merge.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Merges the two given Records, recursively merging any nested Records with
+ * the second collection overriding the first in case of conflict
+ *
+ * For arrays, maps and sets, a merging strategy can be specified to either
+ * "replace" values, or "concat" them instead.
+ * Use "includeNonEnumerable" option to include non enumerable properties too.
+ */
```

Comment:
I think we should include an example, as that really helps to understand the basic idea of the function. See all other functions in the module for examples for...examples.

Reply:
So I included an example, but used `assertEquals` from the testing instead.

What bugs me with the other examples is that some are `console.assert([] === [])` or `console.assert({} === {})` which will always be false since they do not refers to same reference. I know it's for examples, but I think in the future they should be changed as it could be misleading (especially if we later do `deno test --doc`

Reply:
Yes you are absolutely correct, that is still on my todo list and will not dtay that way.